app/services/csv/lettings_log_csv_service.rb

```diff
@@ -7,35 +7,45 @@ def initialize(user)
       set_csv_attributes
     end
 
-    def to_csv
+    def to_csv(is_codes_only_export:)
       CSV.generate(headers: true) do |csv|
         csv << @attributes
 
         LettingsLog.all.find_each do |record|
           csv << @attributes.map do |att|
-            label_from_value(record, att)
+            label_from_value(record, att, is_codes_only_export:)
           end
         end
       end
     end
 
   private
 
-    def label_from_value(record, att)
+    def label_from_value(record, att, is_codes_only_export:)
       if %w[la prevloc].include? att
-        label_from_boolean_value(record.send(att))
+        record.send(att)
       elsif %w[mrcdate startdate voiddate].include? att
         record.send(att)&.to_formatted_s(:govuk_date)
+      elsif is_codes_only_export && att.start_with?("location_", "scheme_")
+        att += "_before_type_cast" unless %w[location_code scheme_code scheme_owning_organisation_name scheme_created_at location_startdate].include? att
+        record.send(att)
       else
-        record.form.get_question(att.remove("_label"), record)&.label_from_value(record.send(att.remove("_label"))) || label_from_boolean_value(record.send(att.remove("_label")))
+        att = att.remove("_label", "_detail") # a couple of csv column headers have suffixes for the user that are not reflected in the app domain
+        field_value = record.send(att)
+        answer_label = record.form
+                             .get_question(att, record)
+                             &.label_from_value(field_value)
```

Comment:
It doesn't look like we need to execute this is it is a codes only export (?) can we bring it down into the if statement after line 40?

Reply:
good spot